src/OSPSuite.Core/Domain/Descriptors/EntityDescriptorMapList.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace OSPSuite.Core.Domain.Descriptors
+{
+   public class EntityDescriptorMapList<T> : List<EntityDescriptorMap<T>> where T : IEntity
+   {
+      public EntityDescriptorMapList()
+      {
+      }
+
+      public EntityDescriptorMapList(IEnumerable<EntityDescriptorMap<T>> collection) : base(collection)
+      {
+      }
+
+      public IReadOnlyList<T> AllSatisfiedBy(DescriptorCriteria criteria)
```

Comment:
The list of such Map. Bonus it returns the entity satisfying a given criteria=> code much simpler